### ISSUE TITLE
Linking with libevent doesn't work out of the box with FreeBSD and OpenIndiana

### ISSUE THREAD

**Issue by rdmark** [Netatalk/netatalk#323]
With FreeBSD's standard clang, and with OpenIndiana's gcc-11, linking with libevent fails when using the default flags.

**Comment by rdmark** [comment]
OpenIndiana 2022-10
```
Making all in netatalk
  CC       netatalk-netatalk.o
  CC       netatalk-afp_avahi.o
  CC       netatalk-afp_mdns.o
  CC       netatalk-afp_zeroconf.o
  CCLD     netatalk
Undefined                       first referenced
 symbol                             in file
event_set_fatal_callback            netatalk-netatalk.o
event_new                           netatalk-netatalk.o
ld: fatal: symbol referencing errors. No output written to .libs/netatalk
collect2: error: ld returned 1 exit status
*** Error code 1
The following command caused the error:
echo "  CCLD    " netatalk;/bin/sh ../../libtool --silent --tag=CC    --mode=link gcc -D_REENTRANT -D_PATH_CONFDIR=\"/usr/local/etc/\"  -D_PATH_STATEDIR='"/usr/local/var/netatalk/"'  -D_PATH_AFPD=\"/usr/local/sbin/afpd\"  -D_PATH_CNID_METAD=\"/usr/local/sbin/cnid_metad\" -I/usr/include/libevent2  -D_U_="__attribute__((unused))" -g -O2 -levent  -o netatalk netatalk-netatalk.o  netatalk-afp_avahi.o netatalk-afp_mdns.o  netatalk-afp_zeroconf.o -lavahi-common -lavahi-client  ../../libatalk/libatalk.la -lsendfile -lsocket -lnsl
make: Fatal error: Command failed for target `netatalk'
Current working directory /export/home/dmark/netatalk/etc/netatalk
*** Error code 1
The following command caused the error:
fail=; \
if (target_option=k; case ${target_option-} in  ?) ;;  *) echo "am__make_running_with_option: internal error: invalid"  "target option '${target_option-}' specified" >&2;  exit 1;;  esac;  has_opt=no;  sane_makeflags=$MAKEFLAGS;  if {  if test -z '2'; then  false;  elif test -n ''; then  true;  elif test -n '' && test -n ''; then  true;  else  false;  fi;  }; then  sane_makeflags=$MFLAGS;  else  case $MAKEFLAGS in  *\\[\ \      ]*)  bs=\\;  sane_makeflags=`printf '%s\n' "$MAKEFLAGS"  | sed "s/$bs$bs[$bs $bs        ]*//g"`;;  esac;  fi;  skip_next=no;  strip_trailopt ()  {  flg=`printf '%s\n' "$flg" | sed "s/$1.*$//"`;  };  for flg in $sane_makeflags; do  test $skip_next = yes && { skip_next=no; continue; };  case $flg in  *=*|--*) continue;;  -*I) strip_trailopt 'I'; skip_next=yes;;  -*I?*) strip_trailopt 'I';;  -*O) strip_trailopt 'O'; skip_next=yes;;  -*O?*) strip_trailopt 'O';;  -*l) strip_trailopt 'l'; skip_next=yes;;  -*l?*) strip_trailopt 'l';;  -[dEDm]) skip_next=yes;;  -[JT]) skip_next=yes;;  esac;  case $flg in  *$target_option*) has_opt=yes; break;;  esac;  done;  test $has_opt = yes); then \
  failcom='fail=yes'; \
else \
  failcom='exit 1'; \
fi; \
dot_seen=no; \
target=`echo all-recursive | sed s/-recursive//`; \
case "all-recursive" in \
  distclean-* | maintainer-clean-*) list='spotlight afpd cnid_dbd netatalk uams' ;; \
  *) list='spotlight afpd cnid_dbd netatalk uams' ;; \
esac; \
for subdir in $list; do \
  echo "Making $target in $subdir"; \
  if test "$subdir" = "."; then \
    dot_seen=yes; \
    local_target="$target-am"; \
  else \
    local_target="$target"; \
  fi; \
  (CDPATH="${ZSH_VERSION+.}:" && cd $subdir && make  $local_target) \
  || eval $failcom; \
done; \
if test "$dot_seen" = "no"; then \
  make  "$target-am" || exit 1; \
fi; test -z "$fail"
make: Fatal error: Command failed for target `all-recursive'
Current working directory /export/home/dmark/netatalk/etc
*** Error code 1
The following command caused the error:
fail=; \
if (target_option=k; case ${target_option-} in  ?) ;;  *) echo "am__make_running_with_option: internal error: invalid"  "target option '${target_option-}' specified" >&2;  exit 1;;  esac;  has_opt=no;  sane_makeflags=$MAKEFLAGS;  if {  if test -z '1'; then  false;  elif test -n ''; then  true;  elif test -n '' && test -n ''; then  true;  else  false;  fi;  }; then  sane_makeflags=$MFLAGS;  else  case $MAKEFLAGS in  *\\[\ \      ]*)  bs=\\;  sane_makeflags=`printf '%s\n' "$MAKEFLAGS"  | sed "s/$bs$bs[$bs $bs        ]*//g"`;;  esac;  fi;  skip_next=no;  strip_trailopt ()  {  flg=`printf '%s\n' "$flg" | sed "s/$1.*$//"`;  };  for flg in $sane_makeflags; do  test $skip_next = yes && { skip_next=no; continue; };  case $flg in  *=*|--*) continue;;  -*I) strip_trailopt 'I'; skip_next=yes;;  -*I?*) strip_trailopt 'I';;  -*O) strip_trailopt 'O'; skip_next=yes;;  -*O?*) strip_trailopt 'O';;  -*l) strip_trailopt 'l'; skip_next=yes;;  -*l?*) strip_trailopt 'l';;  -[dEDm]) skip_next=yes;;  -[JT]) skip_next=yes;;  esac;  case $flg in  *$target_option*) has_opt=yes; break;;  esac;  done;  test $has_opt = yes); then \
  failcom='fail=yes'; \
else \
  failcom='exit 1'; \
fi; \
dot_seen=no; \
target=`echo all-recursive | sed s/-recursive//`; \
case "all-recursive" in \
  distclean-* | maintainer-clean-*) list='include libatalk bin config etc contrib distrib doc man macros test' ;; \
  *) list='include libatalk bin config etc contrib distrib doc man macros test' ;; \
esac; \
for subdir in $list; do \
  echo "Making $target in $subdir"; \
  if test "$subdir" = "."; then \
    dot_seen=yes; \
    local_target="$target-am"; \
  else \
    local_target="$target"; \
  fi; \
  (CDPATH="${ZSH_VERSION+.}:" && cd $subdir && make  $local_target) \
  || eval $failcom; \
done; \
if test "$dot_seen" = "no"; then \
  make  "$target-am" || exit 1; \
fi; test -z "$fail"
make: Fatal error: Command failed for target `all-recursive'
Current working directory /export/home/dmark/netatalk
*** Error code 1
make: Fatal error: Command failed for target `all'
```

**Comment by rdmark** [comment]
FreeBSD 13.1
```
Making all in netatalk
  CC       netatalk-netatalk.o
  CC       netatalk-afp_avahi.o
  CC       netatalk-afp_mdns.o
  CC       netatalk-afp_zeroconf.o
  CCLD     netatalk
ld: error: unable to find library -levent
cc: error: linker command failed with exit code 1 (use -v to see invocation)
*** Error code 1

Stop.
make[3]: stopped in /usr/home/dmark/netatalk/etc/netatalk
*** Error code 1

Stop.
make[2]: stopped in /usr/home/dmark/netatalk/etc
*** Error code 1

Stop.
make[1]: stopped in /usr/home/dmark/netatalk
*** Error code 1

Stop.
make: stopped in /usr/home/dmark/netatalk
```

**Comment by rdmark** [comment]
In the configure summary, it seems like the libevent flag isn't set:
```
    LIBEVENT:
        LIBS   =
        CFLAGS = -I/usr/local/include
```

**Comment by rdmark** [comment]
Ok with FreeBSD this did the trick:
```
LDFLAGS="-L/usr/local/lib" LIBS="-levent" ./configure
```

**Comment by rdmark** [comment]
I think this is the key part of the issue on OpenIndiana:
```
Undefined                       first referenced
 symbol                             in file
event_set_fatal_callback            netatalk-netatalk.o
event_new                           netatalk-netatalk.o
ld: fatal: symbol referencing errors. No output written to .libs/netatalk
collect2: error: ld returned 1 exit status
```

With two symbols that aren't defined, perhaps OpenIndiana's libevent isn't compatible? The libevent provided with OpenIndiana is 2.0 while Debian Bullseye and FreeBSD are using v2.1

**Comment by rdmark** [comment]
So I figured out that it was trying to link against libevent v1. OpenIndiana has both libevent1 and libevent2 packages.

However when installing libevent2 the linker still couldn't find the library, so I figured it was looking for libevent.so and not libevent2.so. So I made a symlink, which made the linker find the library, but then it errored out with `wrong ELF class: ELFCLASS32` which I think means that my 64 bit system doesn't like the 32 bit library.

Ultimately I built my own libevent from the 2.1.12-stable tarball, and all is well now. Will make a wiki page to cover this.
